libs/testproxy/server.go

```diff
@@ -133,15 +138,32 @@ func (s *ProxyServer) proxyToCloud(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	// The SDK reads the X-Databricks-Org-Id header to get the workspace ID. Set it for
+	// the /api/2.0/preview/scim/v2/Me endpoint so that the SDK CurrentWorkspaceID()
+	// method returns the workspace ID.
+	responseHeaders := http.Header{}
+	if r.Method == "GET" && r.URL.Path == "/api/2.0/preview/scim/v2/Me" {
+		workspaceId, err := s.workspaceClient.CurrentWorkspaceID(context.Background())
+		require.NoError(s.t, err)
+		responseHeaders["X-Databricks-Org-Id"] = []string{strconv.FormatInt(workspaceId, 10)}
```

Comment:
should not we just proxy all x- headers?

Reply:
see: https://github.com/databricks/databricks-sdk-go/blob/8a41927168960b20623735caed99a059d53ac0e6/service/files/model.go#L122

Reply:
> can we use standard http client?

No, because we need auth from the SDK client. But the new client is powerful enough. If we wish all x- headers we need a change in the SDK. I'm skipping that for now to avoid scope creep.